src/frontend/Deprecation_analysis.ml

```diff
@@ -85,50 +88,52 @@ let rec collect_deprecated_expr deprecated_userdefined
     ({expr; emeta} : (typed_expr_meta, fun_kind) expr_with) :
     (Location_span.t * string) list =
   match expr with
-  | GetLP ->
-      acc
-      @ [ ( emeta.loc
-          , "The no-argument function `get_lp()` is deprecated. Use the \
-             no-argument function `target()` instead." ) ]
   | FunApp (StanLib FnPlain, {name= "abs"; _}, [e])
     when Middle.UnsizedType.is_real_type e.emeta.type_ ->
       collect_deprecated_expr deprecated_userdefined
         ( acc
         @ [ ( emeta.loc
             , "Use of the `abs` function with real-valued arguments is \
-               deprecated; use functions `fabs` instead." ) ] )
+               deprecated; use function `fabs` instead." ) ] )
         e
   | FunApp (StanLib FnPlain, {name= "if_else"; _}, l) ->
       acc
       @ [ ( emeta.loc
-          , "The function `if_else` is deprecated. Use the conditional \
-             operator (x ? y : z) instead." ) ]
+          , "The function `if_else` is deprecated and will be removed in Stan \
+             2.32.0. Use the conditional operator (x ? y : z) instead; this \
+             can be automatically changed using stanc --print-canonical" ) ]
       @ List.concat
           (List.map l ~f:(fun e ->
                collect_deprecated_expr deprecated_userdefined [] e ) )
   | FunApp (StanLib _, {name; _}, l) ->
       let w =
         if Option.is_some (String.Map.find deprecated_distributions name) then
+          let rename, version =
+            String.Map.find_exn deprecated_distributions name in
```

Comment:
Instead of doing `Map.find` twice you should just `match .. with` the first result. Also I think `Map.find` is preferred over `String.Map.find`.

Reply:
Yeah, the structure of this entire file is really awful for this reason. I was trying to avoid redoing all of it, especially because it will now mostly be deleted a couple versions from now, but it's probably worth it.

Reply:
If you decide rewriting all this is not worth it then I'd still change that `Option.is_some  (String.Map.find ..)` to `Map.mem`